### ISSUE TITLE
🐛 Biome not detected in dependencies

### ISSUE THREAD

**Issue by tylerlaprade** [biomejs/biome-vscode#283]
### VS Code version

1.91.1

### Extension version

2.3.0

### Biome version

1.8.3

### Operating system

- [ ] Windows
- [X] macOS
- [ ] Linux

### Description

I see this popup, and don't get any linting from Biome, even though I have Biome in my project's `devDependencies` in `package.json`

<img width="1031" alt="image" src="https://github.com/user-attachments/assets/26100c6b-2ad3-480d-a3c6-6a916704524e">

### Steps to reproduce

1. Add `"@biomejs/biome": "1.8.3",` to `devDependencies`
2. Install the Biome extension
3. Reload the window

### Expected behavior

Biome should detect that it's installed from `package.json`

### Does this issue occur when using the CLI directly?

Yes

**Comment by ematipico** [comment]
Thank you @tylerlaprade for opening the issue. Would you mind sharing a reproduction?

**Comment by tylerlaprade** [comment]
@ematipico, once I clicked `Download Biome`, I'm not getting it anymore, but that is just a band-aid fix. I tried deleting all "Biome" files on my computer in order to repro, but it's still globally installed somewhere. (Not on the command line, so I can't `which biome`).

To repro, any node project should suffice with Biome in `devDependencies`.

**Comment by ematipico** [comment]
Sorry @tylerlaprade, but I use Biome myself in other Node.js. projects, and I don't experience this issue, it's possible that it's something related to your project or your machine.

**Comment by tylerlaprade** [comment]
Sure, makes sense. In that case, what's the best way for me to uninstall Biome from wherever that button put it? I can't find it in file search.

**Comment by ematipico** [comment]
@tylerlaprade  There's a command palette for that `Biome: clear versions cache`

**Comment by tylerlaprade** [comment]
@ematipico, I still have this "bundled" thing showing after clearing versions cache. My teammate is also getting the constant popup warning as well.

<img width="211" alt="image" src="https://github.com/user-attachments/assets/cdb956d3-924d-4b3b-87f6-ed66bf5bd5f3">

**Comment by nhedger** [comment]
Ah, this might have been poor communication on my part. This command does not actually clear the downloaded versions, but we cache the request that retrieves the list of available versions from GitHub to avoid hitting the rate limit. This command is used to reset this cache.

To clear this "bundled" version, you need to delete the following folder manually:

```
~/Library/Application Support/Code/User/globalStorage/biomejs.biome/server
```

We'll make sure to add a command that does this in the next version.

**Comment by nhedger** [comment]
I'm closing this as I believe the latest nightly fixes this issue. Let me know if it's not the case.

**Comment by tylerlaprade** [comment]
Thanks @nhedger. I assume you meant pre-release, as "nightly" is deprecated.
<img width="379" alt="image" src="https://github.com/user-attachments/assets/8ece6281-aa1a-4c50-abef-7808c7543bbe">
After switching to pre-release, I have a _different_ popup, but I'm still missing Biome. I expect to have it because it's in my `package.json` and I've run `npm i`. Could the problem be that it's a monorepo and `package.json` isn't in the root folder?
<img width="493" alt="image" src="https://github.com/user-attachments/assets/58419d85-80c9-4662-b625-2da7c2255de6">

**Comment by tylerlaprade** [comment]
@nhedger, hmm, that's still not working, both with and without a slash at the end of the path. I'm in a monorepo, but with only one relevant project - the back-end is in Python. So I have `biome.jsonc` in my workspace root, but `package.json` and `node_modules/` are in the subfolder. I confirmed that `node_modules/` contains `@biomejs`.
<img width="400" alt="image" src="https://github.com/user-attachments/assets/bd475bd2-74de-4380-a16f-751fcbb14186">
<img width="328" alt="image" src="https://github.com/user-attachments/assets/eee19cc2-4cec-4b05-93a5-58d6bf71fc5b">

**Comment by nhedger** [comment]
Would you mind providing a minimal reproduction repo? I'll have a look!

**Comment by tylerlaprade** [comment]
Sure! I was able to repro without a monorepo.
https://github.com/tylerlaprade/biome_bug
**Steps:**
1. Open this project in a fresh VSCode install
2. Run `npm i` to install Biome
3. `Developer: Reload Window` to see the error trigger

**Comment by tylerlaprade** [comment]
Sure, how's this: https://www.loom.com/share/708aae2faab4444b8a31caf6e0931ba3?sid=833839d4-f0d1-49f3-a8e7-6667ac0782f4

**Comment by nhedger** [comment]
I think I found the issue.

When the extension boots, it creates a global LSP session and a project-specific LSP session. I believe the popup you see is triggered by the global LSP session starting because a global Biome cannot be found on your system. When you install Biome in your node_modules and restart VS Code, it again attempts to start the global session. When you dismiss the notification, it actually boots the project-specific LSP session (you can see Biome 1.8.3) in your status bar.

I initially intended for users to download Biome on the fly using the popup so that they could start the global LSP session. However, I'm realizing now that the intent may not be obvious.

In the meantime, clicking `Yes` in the popup to download Biome should solve the issue. This will download a biome binary used for the global LSP session and shouldn't prompt you anymore.

I'll have to think of a better way to signal users why the see this popup.

**Comment by tylerlaprade** [comment]
@nhedger, even though I see "Biome 1.8.3" in the status bar, the extension doesn't seem to be able to make use of it. I don't get formatting on save or any errors/warning squiggles in my code

Regarding clicking `Yes` for the global download, I don't want my local environment to ever use a different version that what I have in CI. That would cause me to not realize my PR will fail the build until I submit it.

**Comment by nhedger** [comment]
To be clear, the downloaded version would only be used with the global session which handles files that are not part of your project such as Untitled (unsaved) files.

**Comment by tylerlaprade** [comment]
Ah okay - in that case it seems like there's still a bug? The project one is not working on files that are part of my project.

**Comment by 321ckatz123** [comment]
@nhedger just for clarity sake, can this be re-opened as it was closed due to belief it was fixed in the pre-release, which doesn't appear to be true?

**Comment by nhedger** [comment]
> @nhedger just for clarity sake, can this be re-opened as it was closed due to belief it was fixed in the pre-release, which doesn't appear to be true?

Would you mind creating a new issue with your own reproduction side, please?

**Comment by tylerlaprade** [comment]
FWIW @321ckatz123 works with me so we both have the same repro :)

**Comment by tylerlaprade** [comment]
@nhedger, I added the following line to my `settings.json` and now I get the original (but still buggy) behavior I first reported.
`"biome.projects": [{"path": "paqarina"}]`
 I now have the correct Biome version displayed in the bottom right, but I am still unable to format or lint any files in VSCode, regardless of whether they're in the project or not. I also get this popup for every single file I open (even though those files are in the `paqarina/` folder), which is quite noisy.
<img width="491" alt="image" src="https://github.com/user-attachments/assets/1140ea3f-6eac-41ed-9a3c-7875ad174374">
<img width="345" alt="image" src="https://github.com/user-attachments/assets/bc334a31-c934-47c7-85ce-9d80e81ee125">

**Comment by nhedger** [comment]
Happy to try and reproduce it if you can provide a minimal reproduction repo with these settings.

**Comment by tylerlaprade** [comment]
Sure thing - here it is @nhedger 
https://github.com/tylerlaprade/biome_bug

**Comment by alepell** [comment]
hello guys, any1 got a fix for this?

**Comment by cimchd** [comment]
Same problem in a yarn 3.2.2 monorepo. 
In the root package.json is a devDependency for @biomejs/biome v1.9.4

But I always get the message `Could not find Biome in your dependencies. Either add the @biomejs/biome package to your dependencies, or download the Biome binary.`

Formatter does not work after I click OK. At the moment only the biome cli is working for me.

I am confused. Is this a bug?, Did I miss something? Or will this only work after https://github.com/biomejs/biome/issues/2228 is implemented?

**Comment by tjb1019** [comment]
I ran into the same issue and was able to get it working by turning off PATH search and manually pointing to the biome bin in node_modules.

`.vscode/settings.json`
```json
{
    "biome.searchInPath": false,
    "biome.lspBin": "client/node_modules/@biomejs/cli-darwin-x64/biome",
}
```

**Comment by tylerlaprade** [comment]
@tjb1019 , does that fix it even for files outside the workspace?

**Comment by jojungdfw** [comment]
> I ran into the same issue and was able to get it working by turning off PATH search and manually pointing to the biome bin in node_modules.
> 
> `.vscode/settings.json`
> 
> {
>     "biome.searchInPath": false,
>     "biome.lspBin": "client/node_modules/@biomejs/cli-darwin-x64/biome",
> }

This worked for me!

**Comment by xinaesthete** [comment]
The above also worked for me - but I find it less than ideal that this necessary. I don't think there's anything particularly complex about my project from a biome perspective (using npm, not a monorepo...)

**Comment by xinaesthete** [comment]
I'm trying to make a repro for this because it's really irritating.

I realised that in this instance, it was in fact not the extension specifically, but that biome was indeed not properly installed.

I think that what happened was that when merging changes that involve changes to `package-lock.json`, I removed it and recreated it by running `npm i` again and committing that version - but this has a different result than the original `npm install --save-dev --save-exact @biomejs/biome`... Having re-run that, it seems to be improved.
